homeassistant/components/amcrest/__init__.py

```diff
@@ -143,7 +153,19 @@ def setup(hass, config):
                 CONF_NAME: name,
             }, config)
 
+        if binary_sensors:
+            discovery.load_platform(
+                hass, 'binary_sensor', DOMAIN, {
+                    CONF_NAME: name,
+                    CONF_BINARY_SENSORS: binary_sensors
+                }, config)
+
         if sensors:
+            if 'motion_detector' in sensors:
+                _LOGGER.warning(
```

Comment:
I will suggest move warning to config schema, see `cv. deprecated()`

Reply:
I tried to see how other, existing code handled deprecation, and I did find and look at cv.deprecated(). However, I'm not sure how that could be used here. It looks like it is designed for config keys, and this is not a config key, but rather one of the choices for the list which is a key's data. If you can show me how to use cv.deprecated() here, I'd be happy to do so, but I couldn't see how it would work in this situation.

Reply:
It could be extended to support only validate specific value, but your current implement is okay.